main.go

```diff
@@ -1,7 +1,51 @@
 package main
 
-import "fmt"
+import (
+	"encoding/json"
+	"fmt"
+	"reflect"
+)
+
+type Domicile struct {
+	country   string `json:"country"`
+	is_remote bool   `json:"is_remote"`
+}
+
+type Employee struct {
+	name            string   `json:"name"`
+	entity          string   `json:"entity"`
+	employee_number int      `json:"employee_number"`
+	salary          float64  `json:"salary"`
+	domicile        Domicile `json:"domicile"`
+}
 
 func main() {
-	fmt.Printf("Hello, world or 你好，世界 or Καλημέρα κόσμε or こんにちは世界\n")
-}
\ No newline at end of file
+	// TODO: task #1 - give me a skeleton!
+	//missing comma after salary
+	data := string(`
+    {
+        "name": "Golang",
+        "entity": "Xendit",
+        "employee_number": 10,
+        "salary": 1.5,
+        "domicile": {
+            "country": "ID",
+            "is_remote": true
+        }
+    }
+    `)
+	var employee Employee
+	if err := json.Unmarshal([]byte(data), &employee); err != nil {
+		fmt.Println("Task #1 failed!")
+		return
+	}
+	// TODO: task #2 - I am a legal employee, include me into your database!
+	var database map[string]Employee
+	database = make(map[string]Employee) //use make map into employee
+	database["Golang"] = employee        // assign employee with "Golang" Key hash map
```

Comment:
now you learn about simple hashmap as well. Good job!

Now you don't need to use Redis for caching purpose. If possible, you could use Golang on here for fast retrieval